src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -454,41 +451,46 @@ private void running2RunningFailed(Task task) {
         }
 
         final List<Replicate> replicates = replicatesService.getReplicates(task.getChainTaskId());
-
         final List<Worker> aliveWorkers = workerService.getAliveWorkers();
 
-        // If an alive worker has not run the task, it is not a `RUNNING_FAILURE`.
-        final Predicate<Worker> hasWorkerAlreadyTried = worker -> replicates
+        final Function<Worker, Optional<Replicate>> getReplicateForWorker = worker -> replicates
```

Comment:
Much easier to read!
We now have some code duplicate can't we benefit of this existing method https://github.com/iExecBlockchainComputing/iexec-core/blob/98171a1759c18b0c8bac81f0a2490e0d634878c7/src/main/java/com/iexec/core/replicate/ReplicatesList.java#L56 ?

Reply:
Sure. I didn't know about this function!